siliconcompiler/sphinx_ext/dynamicgen.py

```diff
@@ -775,16 +775,47 @@ def document_module(self, module, modname, path):
         return section
 
 def keypath_role(name, rawtext, text, lineno, inliner, options=None, content=None):
+    doc = inliner.document
+    env = doc.settings.env
+
     # Split and clean up keypath
     keys = [key.strip() for key in text.split(',')]
     try:
-        return [keypath(*keys)], []
+        return [keypath(keys, env.docname)], []
     except ValueError as e:
         msg = inliner.reporter.error(f'{rawtext}: {e}', line=lineno)
         prb = inliner.problematic(rawtext, rawtext, msg)
         return [prb], [msg]
 
+class SCDomain(sphinx.domains.std.StandardDomain):
+    name = 'sc'
+
+    # Override in StandardDomain so xref is literal instead of inline
+    def build_reference_node(self, fromdocname, builder, docname, labelid, sectname, rolename, **options):
```

Comment:
Is the body of this function derived from the original in StandardDomain? Maybe include link to the source if so.

Reply:
Good point